File: TestStreamingApiCSharp/FSharpExtensions.cs
using Microsoft.FSharp.Control;
using Microsoft.FSharp.Core;
using System.Threading;
using System.Threading.Tasks;

namespace BeloSoft.Extensions
{
    public static class FSharpExtensions
    {
        public static Task<T> ExecuteAsyncTask<T>(this FSharpAsync<T> task)
        {
            return FSharpAsync.StartAsTask(task, FSharpOption<TaskCreationOptions>.None, FSharpOption<CancellationToken>.None);
        }
    }
}

File: TestStreamingApiVB/FSharpExtensions.vb
Imports System.Runtime.CompilerServices
Imports System.Threading
Imports Microsoft.FSharp.Control
Imports Microsoft.FSharp.Core

Module FSharpExtensions

    <Extension>
    Public Function ExecuteAsyncTask(Of T)(task As FSharpAsync(Of T)) As Task(Of T)
        Return FSharpAsync.StartAsTask(task, FSharpOption(Of TaskCreationOptions).None, FSharpOption(Of CancellationToken).None)
    End Function
End Module